assets/js/page-enhancements.js

```diff
@@ -80,17 +78,77 @@
           navigator.clipboard.write([new ClipboardItem({ 'image/png': blob })])
         }, 'image/png')
       })
-  })
-})($, document)
-;(autoDownloadCoversheetPdf = ($, document) => {
-  $(document).ready(() => {
+  }
+
+  const autoDownloadCoversheetPdf = () => {
     setTimeout(() => {
       $('#print-coversheets-content #downloadPdf').each((idx, element) => element.click())
     }, 1000)
-  })
-})($, document)
-;(autoFocusBarcodeField = ($, document) => {
-  $(document).ready(() => {
-    $('#scan-barcode-form #barcode').focus()
-  })
+  }
+
+  const autoFocusBarcodeField = () => {
+    $('#scan-barcode-form #barcode').trigger('focus')
+  }
+
+  const extractCsrfFieldAsKeyValuePairFromForm = form => {
+    return form
+      .serializeArray()
+      .filter(formField => formField.name === '_csrf')
+      .map(formField => {
+        return `${formField.name}=${formField.value}`
+      })
+  }
+
+  const submitCookiePreferencesFormViaAjax = () => {
+    $('#cookiePreferencesForm button[type=submit]').on('click', function (event) {
+      event.preventDefault()
+      const form = $(this).closest('form')
+      const csrfQueryString = extractCsrfFieldAsKeyValuePairFromForm(form)
```

Comment:
Because its an ajax request (appliction/json), traditional form fields cannot be sent in the same way as a browser POST request does. In respect of the `_csrf` field we need to extract it and add it as a query string parameter. All other fields/data (if there are any) can be serialized into an object and sent as the request body